src/main/kotlin/com/exactpro/th2/ws/client/api/impl/WebSocketClient.kt

```diff
@@ -77,6 +81,7 @@ class WebSocketClient(
 
     override fun onOpen(socket: WebSocket) = try {
         onInfo { "Connected to: $uri" }
+        this.socket = socket
```

Comment:
Just to understand the problem: what was the cause of the deadlock? The volatile field?
When I looked into the `buildAsync` method  I did not see any code that waits for `onOpen` method is invoked

Reply:
Seems like future returned from `buildAsync` will complete only when `onOpen` is completed. And if we call `start` from one thread and then try to send smth from `onOpen` we're gonna get deadlocked since both start and send are called from different threads and try to acquire the same lock

Reply:
Yep, I and @Topru333  thought the same way. But I didn't find the exact place where it's happening